core/trino-main/src/main/java/io/trino/operator/MultiChannelGroupByHash.java

```diff
@@ -219,6 +223,9 @@ public void appendValuesTo(int groupId, PageBuilder pageBuilder, int outputChann
         if (isRunLengthEncoded(page)) {
```

Comment:
you should call `page = page.getLoadedPage()` before dictionary checks (same for `getGroupIds`). Same for `BigintGroupByHash`